### PR TITLE
Fix recently found bugs

### PR DESCRIPTION
**Pull request by alkurbatov**
* Document how to launch old replays (OS X + Win).
* Split the input params with '--'. The observer's params go before '--', the game params go after '--'.
* Rebase on the newer API: Abort the game launch if we found an unexpected option.
* Integrate Google tests.

### CODE REVIEW

**Review by lladdy — APPROVED** [review]
Looks good to me.
Does this need any further testing?

**Comment by alkurbatov** [comment]
Small refactoring needed for better testing later and there is missing delimiter in the windows example.

**Comment by alkurbatov** [comment]
Changes:
* Moved the input args split function to separate files in order to reuse for testing later.
* Add missing delimiter to the example.

Not checked:
* Windows compilation.

**Comment by alkurbatov** [comment]
Now with unit tests :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.10)
```
Comment (alkurbatov) [inline CMakeLists.txt:1]: Required to use `gtest_discover_tests`

**Pull request merged**